src/main/java/com/maxmind/db/Networks.java

```diff
@@ -0,0 +1,184 @@
+package com.maxmind.db;
+
+import java.io.IOException;
+import java.net.Inet4Address;
+import java.net.InetAddress;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.Stack;
+
+/**
+ * Instances of this class provide an iterator over the networks in a database.
+ * The iterator will return a {@link DatabaseRecord} for each network.
+ * 
+ * @param <T> The type of data returned by the iterator.
+ */
+public class Networks<T> implements Iterator<DatabaseRecord<T>> {
+    private final Reader reader;
+    private final Stack<NetworkNode> nodes;
+    private NetworkNode lastNode;
+    private final boolean skipAliasedNetworks;
+    private final ByteBuffer buffer; /* Stores the buffer for Next() calls */
+    private Class<T> typeParameterClass;
+    
+    /**
+     * Constructs a Networks instance.
+     * @param reader The reader object.
+     * @param skipAliasedNetworks The boolean to skip aliased networks.
+     * @throws ClosedDatabaseException Exception for a closed database.
+     */
+    Networks(Reader reader, boolean skipAliasedNetworks) 
+        throws ClosedDatabaseException {
+        this(reader, skipAliasedNetworks, new NetworkNode[]{});
+    }
+
+    /**
+     * Constructs a Networks instance.
+     * @param reader The reader object.
+     * @param skipAliasedNetworks The boolean to skip aliased networks.
+     * @param nodes The initial nodes array to start Networks iterator with.
+     * @throws ClosedDatabaseException Exception for a closed database.
+     */
+    Networks(Reader reader, boolean skipAliasedNetworks, NetworkNode[] nodes) 
+        throws ClosedDatabaseException {
+        this.reader = reader;
+        this.skipAliasedNetworks = skipAliasedNetworks;
+        this.buffer = reader.getBufferHolder().get();
+        this.nodes = new Stack<NetworkNode>();
+        for (NetworkNode node : nodes) {
+            this.nodes.push(node);
+        }
+    }
+
+    /**
+     * Constructs a Networks instance with skipAliasedNetworks set to false by default.
+     * @param reader The reader object.
+     */
+    Networks(Reader reader) throws ClosedDatabaseException {
+        this(reader, false);
+    }
+
+    /**
+     * Sets the Class for the data type in DataRecord.
+     * @param cls The class object. ( For example, Map.class )
+     */
+    public void setDataClass(Class<T> cls) {
```

Comment:
Is there any reason why this can't just be part of the constructor?

If we do need it, it should be package private, but I don't think we need it.

Reply:
The code in go lets people retrieve data into different types when `network(result any)` is called. I thought we would want keep that. But, I agree. I'll move it to the constructor.

Reply:
I don't think that would have worked here anyway given that the class `Networks<T>` has a type parameter and the `T` here needs to match that.